hoomd/state.py

```diff
@@ -121,8 +143,45 @@ def snapshot(self, snapshot):
 
         self._cpp_sys_def.initializeFromSnapshot(snapshot._cpp_obj)
 
+    @property
+    def particle_types(self):
+        """list[str]: List of all particle types in the simulation."""
```

Comment:
```suggestion
        """List[str]: List of all particle types in the simulation."""
```
I thought we were using the Python typing notation. Should we mention that these are all copies and are not settable?

Reply:
I was until I noticed that it wouldn't link to the python standard documentation. We should mention that. I think the type documentation could be a good point of discussion.

Reply:
`list[str]` links only because Sphinx has hardcoded this for list, dict, and tuple: https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#info-field-lists

We should discuss and decide on a standard to use. If links to `typing.List` are desired, then we can explicitly include those. I have no particular preference as long as we are consistent and clear.

Reply:
This is an annoying inconsistency between Sphinx and Python typing. 😕 I want to recommend `list` over `List`, based on how the output looks. The link to the Python docs for `list` is what I would expect here.